src/utils/utilities.ts

```diff
@@ -240,3 +244,189 @@ export const calculateTimeLeft = (createdAt: string) => {
+            signerMatched = true;
+          }
+
+          // const val = WalletUtilities.getFingerprintFromExtendedKey(
+          //   derivation.pubkey.toString('hex'),
+          //   network
+          // );
+        });
+      }
+    });
+
+    // Check if signer exists in the psbt
+    const currentSignerFromPSBT = signersList.find(
+      (psbtSigner) => psbtSigner.masterFingerprint === signer.masterFingerprint.toLowerCase()
+    );
+
+    // Extract input addresses
+    const senderAddresses = psbt.txInputs.map((input) => input.hash.toString('hex'));
+
+    // Extract outputs (receiver information)
+    const outputs = psbt.txOutputs.map((output) => {
+      return {
+        address: bitcoin.address.fromOutputScript(
+          output.script,
+          isTestnet() ? bitcoin.networks.testnet : bitcoin.networks.bitcoin
+        ), // Receiver address
+        amount: output.value, // Amount in satoshis
+      };
+    });
+
+    // Calculate the total input and output amounts
+    let totalInput = 0;
+    let totalOutput = 0;
+    let totalAmount = 0;
+
+    psbt.data.inputs.forEach((input, index) => {
+      if (input.witnessUtxo) {
+        totalInput += input.witnessUtxo.value;
+      } else if (input.nonWitnessUtxo) {
+        const tx = bitcoin.Transaction.fromBuffer(input.nonWitnessUtxo);
+        const voutIndex = psbt.txInputs[index].index;
+        totalInput += tx.outs[voutIndex].value;
+      }
+    });
+
+    outputs.forEach((output) => {
+      totalOutput += output.amount;
+      if (!changeAddress.includes(output.address)) {
+        totalAmount += output.amount;
+      }
+    });
+
+    const receiverAddresses = outputs.map((op) => op.address);
+
+    // Calculate transaction fees
+    const fees = totalInput - totalOutput;
+    const feeRate = (fees / vBytes).toFixed(2);
+    return {
+      senderAddresses: senderAddresses,
+      receiverAddresses: receiverAddresses,
+      fees,
+      sendAmount: totalAmount,
+      currentSignerFromPSBT,
+      signerMatched,
+      changeAddress,
+      txId,
+      vout,
+      feeRate,
+      vBytes,
+    };
+  } catch (error) {
+    console.log('🚀 ~ dataFromPSBT ~ error:', error);
+    throw 'Something went wrong';
+  }
+};
+
+export const estimateVByteFromPSBT = (base64Str: string) => {
+  const psbt = bitcoin.Psbt.fromBase64(base64Str);
+  const unsignedTxHex =
+    psbt.txInputs.length > 0 && psbt.txOutputs.length > 0 ? psbt.__CACHE.__TX.toHex() : null;
+  const tx = bitcoin.Transaction.fromHex(unsignedTxHex);
+
+  // Calculate the base size (without witness data)
+  const baseSize = tx.toBuffer().length;
+
+  // Check if there is witness data and calculate the total size accordingly
+  const totalSize = tx.hasWitnesses() ? tx.virtualSize() : baseSize;
+
+  // Calculate vBytes using the formula
+  return Math.ceil((baseSize * 3 + totalSize) / 4);
+};
+
+export const getTnxDetailsPSBT = (averageTxFees, feeRate: string) => {
```

Comment:
This returns fee details, should have a name indicating that clearly.

Reply:
its clear from the function name (estimateVByteFromPSBT)

Reply:
The name is getTxnDetailsPSBT

Reply:
getTxnDetailsPSBT returns estimatedBlocksBeforeConfirmation, tnxPriority 
so that's clear with named exports.

Reply:
Those are all fee related only. So it doesn't return transaction details, only fee realted details, which is why it's confusing

Reply:
@ben-kaufman 
will be updating this with the implementation of new screens where more parameters might get added aswell